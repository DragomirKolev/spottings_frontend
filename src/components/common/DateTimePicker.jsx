import React from 'react';
import DateTimePicker from 'react-datetime-picker';
import { createUseStyles } from 'react-jss';

import { GLOBAL_SPACING, styles } from '../../styles/common';

const useStyles = createUseStyles({
  container: {
    marginTop: GLOBAL_SPACING,
    paddingBottom: GLOBAL_SPACING,
    borderBottom: styles.border('green'),
  },
  picker: { width: '100%' }
});

const CustomDateTimePicker = ({ value, onChange }) => {
  const classes = useStyles();

  return (
    <div className={classes.container}>
      <DateTimePicker
        className={classes.picker}
        onChange={onChange}
        value={value}
      />
    </div>
  );
};

export default CustomDateTimePicker;