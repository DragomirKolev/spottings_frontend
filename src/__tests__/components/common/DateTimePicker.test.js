import React from 'react';
import renderer from 'react-test-renderer';

import DateTimePicker from '../../../components/common/DateTimePicker';

it('renders DateTimePicker', () => {
  const tree = renderer
    .create(<DateTimePicker 
          value={'11/11/1111 11:11'} 
          onChange={jest.fn()}
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
  

