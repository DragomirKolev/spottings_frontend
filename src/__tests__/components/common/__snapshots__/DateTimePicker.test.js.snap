// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DateTimePicker 1`] = `
<div
  className="container-0-2-1"
>
  <div
    className="react-datetime-picker react-datetime-picker--closed react-datetime-picker--enabled picker-0-2-2"
    onFocus={[Function]}
  >
    <div
      className="react-datetime-picker__wrapper"
    >
      <div
        className="react-datetime-picker__inputGroup"
        onClick={[Function]}
      >
        <input
          max="275760-09-13T02:00"
          min="-271821-04-20T00:19"
          name="datetime"
          onChange={[Function]}
          onFocus={[Function]}
          step={60}
          style={
            Object {
              "left": "-9999px",
              "position": "absolute",
              "top": "-9999px",
              "visibility": "hidden",
            }
          }
          type="datetime-local"
          value="1111-11-11T11:11"
        />
        
        <input
          autoComplete="off"
          className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__month"
          max={12}
          min={1}
          name="month"
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          placeholder="--"
          required={null}
          type="number"
          value={11}
        />
        <span
          className="react-datetime-picker__inputGroup__divider"
        >
          /
        </span>
        <input
          autoComplete="off"
          autoFocus={false}
          className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__day"
          max={30}
          min={1}
          name="day"
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          placeholder="--"
          required={null}
          type="number"
          value={11}
        />
        <span
          className="react-datetime-picker__inputGroup__divider"
        >
          /
        </span>
        <input
          autoComplete="off"
          autoFocus={false}
          className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__year"
          max={275760}
          min={1000}
          name="year"
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          placeholder="----"
          required={null}
          step={1}
          type="number"
          value={1111}
        />
        <span
          className="react-datetime-picker__inputGroup__divider"
        >
          Â 
        </span>
        <input
          autoComplete="off"
          autoFocus={false}
          className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__hour"
          max={12}
          min={1}
          name="hour12"
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          placeholder="--"
          required={null}
          type="number"
          value={11}
        />
        <span
          className="react-datetime-picker__inputGroup__divider"
        >
          :
        </span>
        <input
          autoComplete="off"
          autoFocus={false}
          className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__minute"
          max={59}
          min={0}
          name="minute"
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          placeholder="--"
          required={null}
          type="number"
          value={11}
        />
        <span
          className="react-datetime-picker__inputGroup__divider"
        >
           
        </span>
        <select
          className="react-datetime-picker__inputGroup__input react-datetime-picker__inputGroup__amPm"
          name="amPm"
          onChange={[Function]}
          required={null}
          value="am"
        >
          <option
            value="am"
          >
            AM
          </option>
          <option
            value="pm"
          >
            PM
          </option>
        </select>
        
      </div>
      <button
        className="react-datetime-picker__clear-button react-datetime-picker__button"
        onClick={[Function]}
        onFocus={[Function]}
        type="button"
      >
        <svg
          className="react-datetime-picker__clear-button__icon react-datetime-picker__button__icon"
          height={19}
          stroke="black"
          strokeWidth={2}
          viewBox="0 0 19 19"
          width={19}
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="4"
            x2="15"
            y1="4"
            y2="15"
          />
          <line
            x1="15"
            x2="4"
            y1="4"
            y2="15"
          />
        </svg>
      </button>
      <button
        className="react-datetime-picker__calendar-button react-datetime-picker__button"
        onClick={[Function]}
        onFocus={[Function]}
        type="button"
      >
        <svg
          className="react-datetime-picker__calendar-button__icon react-datetime-picker__button__icon"
          height={19}
          stroke="black"
          strokeWidth={2}
          viewBox="0 0 19 19"
          width={19}
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="15"
            width="15"
            x="2"
            y="2"
          />
          <line
            x1="6"
            x2="6"
            y1="0"
            y2="4"
          />
          <line
            x1="13"
            x2="13"
            y1="0"
            y2="4"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
